### PR TITLE
Add component subaccount list

### LINKED ISSUE #204: Sub accounts list

### PR DESCRIPTION
**Pull request by colganmeanor**
### Add subaccounts list component

Links
-----

Closes #204  

Developer considerations
--------------

- [ ] Increment the version number for relevant packages
  - [ ] This is a new MAJOR version (incompatible API changes were made)
  - [ ] This is a new MINOR version (functionality was added in a backwards compatible manner)
  - [ ] This is a new PATCH version (backwards compatible bug fixes)
- [ ] Update the CHANGELOG.md with a summary of the changes
  - If this is a MAJOR version, what are the breaking API changes? Was anything added to the API?
  - If this is a MINOR version, what feature was added? Why would consumers want this new version?
  - If this is a PATCH version, what was fixed?
- [ ] Update the appropriate documentation. Keep in mind that this mono repository has README files at the repository, project, and component levels. We want developers to be able to find what they need from the main README, so make sure to add links to component documentation if adding something new.

Steps for Testing/QA
--------------------

TBD

### CODE REVIEW

File: stencil-library/src/api/SubAccount.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  created_at: string,
+  currency: string,
+  id: string,
+  rate_type: string,
+  transaction_fee: number,
+  updated_at: string
+}
+
+export interface ISubAccount {
+  account_type: string,
```
Comment (TiagoFuelber) [inline stencil-library/src/api/SubAccount.ts:62]: @colganmeanor could this be an enum?

File: stencil-library/src/api/SubAccount.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  platform_account_id: string,
+  processing_ready: boolean,
+  status: SubAccountStatusType,
+  updated_at: string
+  related_accounts?: object,
+  readonly statusTagType?: TagTypes
+};
+
+export class SubAccount implements ISubAccount {
+  public account_type: string;
```
Comment (TiagoFuelber) [inline stencil-library/src/api/SubAccount.ts:78]: @colganmeanor also this one, should it be an enum maybe?

File: stencil-library/src/api/SubAccount.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  public timeString: string;
+  public currency: string;
+  public id: string;
+  public name: string;
+  public payout_ready: boolean;
+  public platform_account_id: string;
+  public processing_ready: boolean;
+  public status: SubAccountStatusType;
+  public updated_at: string;
+  public related_accounts: object;
```
Comment (TiagoFuelber) [inline stencil-library/src/api/SubAccount.ts:91]: @colganmeanor this one sounds like it was going to be an array of accounts

File: stencil-library/src/api/SubAccount.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    const activeAchRate = this.findActiveRate(subAccount, 'ach');
+    this.cardRate = this.formattedRate(activeCardRate);
+    this.achRate = this.formattedRate(activeAchRate);
+  }
+
+  findActiveRate = (subAccount: ISubAccount, type: string): ApplicationFeeRate | undefined => {
+    return subAccount.application_fee_rates.find(rate => rate.rate_type === type);
+  };
+
+  basisPercentage = (rate: ApplicationFeeRate | undefined): string | undefined => {
```
Comment (TiagoFuelber) [inline stencil-library/src/api/SubAccount.ts:120]: @colganmeanor it's good practice to name functions and methods with verbs. For example `getBasisPercentage`. `basisPercentage` sounds more like just a variable.

File: stencil-library/src/api/SubAccount.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    return rate && formatPercentage(rate.basis_point_rate);;
```
Comment (TiagoFuelber) [inline stencil-library/src/api/SubAccount.ts:121]: It's ok to check the rate before passing it, but you could also use optional chaining `formatPercentage(rate?.basis_point_rate)` which would look a bit more concise and modern. Also there is an extra colon in the end `;;`

File: stencil-library/src/components/subaccounts-list/subaccounts-list.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  async fetchData(direction?: string): Promise<void> {
+    if (!this.accountId || !this.authToken) {
+      this.errorMessage = "Can not fetch any data without an AccountID and an AuthToken";
+      this.loading = false;
+      return;
+    }
+    this.loading = true;
+    const endpoint = `account/${this.accountId}/seller_accounts`;
+
+    const response: IApiResponseCollection<SubAccount[]> = await Api(this.authToken, 'http://localhost:3000').get(endpoint, {
```
Comment (TiagoFuelber) [inline stencil-library/src/components/subaccounts-list/subaccounts-list.tsx:50]: I think we will probably need to get this value `localhost:3000` from an `.env` file.

Comment (colganmeanor) [reply stencil-library/src/components/subaccounts-list/subaccounts-list.tsx:50]: Yes, good catch. Thanks Tiago I totally forgot about that. 

I think you're right, we should merge the other branch with the .env file before merging this one and that should make this an easy change on this branch once that is completed.

File: stencil-library/src/api/SubAccount.ts (hunk 4 above)
Comment (colganmeanor) [reply stencil-library/src/api/SubAccount.ts:120]: Good call - thank you! Fixing this for sure

File: stencil-library/src/api/SubAccount.ts (hunk 5 above)
Comment (colganmeanor) [reply stencil-library/src/api/SubAccount.ts:121]: Good catch!

File: stencil-library/src/api/SubAccount.ts (hunk 3 above)
Comment (colganmeanor) [reply stencil-library/src/api/SubAccount.ts:91]: So this was taken directly from the type declaration in customer-admin-fe - I agree that the variable name sounds that way but it appears to be typed as an `object`  in our API Docs and thats the way it is in the customer admin as well. 🤷🏻

File: stencil-library/src/api/SubAccount.ts (hunk 1 above)
Comment (colganmeanor) [reply stencil-library/src/api/SubAccount.ts:62]: Sure - easy enough to add! Good call

**Review by TiagoFuelber — APPROVED** [review]
Nice work! Looks good to me

**Review by jakemer — APPROVED** [review]

**Pull request merged**